addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java

```diff
@@ -124,4 +126,26 @@ public static boolean hasSiteStats(String site, String prefix) {
         }
         return list;
     }
+
+    /** A method which mimics the escaping used for traditional ZAP reports */
+    public String replaceEntity(String text) {
+        String result = text;
+        if (text != null) {
+            result = StringEscapeUtils.escapeXml10(XMLStringUtil.escapeControlChrs(result));
+        }
+        return result;
```

Comment:
Do away with the intermediate storage?
```java
            return text != null ? StringEscapeUtils.escapeXml10(XMLStringUtil.escapeControlChrs(text)) : text;
```

Reply:
Actually both of those methods handle null so no need to even do the null test, so updated to remove that :)